test/decode.js

```diff
@@ -43,7 +59,6 @@ describe('decode', function() {
 
   it('throws when URI does not contain comma', function() {
     expect(function() { dataUri.decode('data:text/plain'); })
-      .to.throw(/^Invalid data URI/);
+      .to.throw(/^Not a data URI/);
```

Comment:
There aren't separate error conditions anymore.  There's just URIs that don't match the regex.  If it's important to you to catch `foo:` I can do that by checking for `data:` first, making the function a little slower.